### PR TITLE
fix(button): improve button accessibility (VIV-1505)

### PR DESCRIPTION
**Pull request by RichardHelm**
Most buttons are announced as button now:

* `<vwc-button label="Click me"></vwc-button>` -> "Click me, button"
* `<vwc-button icon="icon" label="Click me"></vwc-button>` -> "Click me, button"
* `<vwc-button icon="icon" title="Click me"></vwc-button>` -> "Click me, button"

However, when using icon button with `aria-label` it is still announced as group:

* `<vwc-button icon="icon" aria-label="Click me"></vwc-button>` -> "Click me, group"

I don't think this can be easily fixed due to how our button works.

### CODE REVIEW

File: libs/components/src/lib/button/button.template.ts [hunk 1]
```diff
@@ -60,39 +60,41 @@ export const buttonTemplate: (
...
+				aria-label="${(x) => x.ariaLabel}"
+				aria-live="${(x) => x.ariaLive}"
+				aria-pressed="${(x) => x.ariaPressed}"
+				aria-relevant="${(x) => x.ariaRelevant}"
+				aria-roledescription="${(x) => x.ariaRoledescription}"
+				title="${x => x.title}"
+				${ref('control')}
+			>
+				${x => renderIconOrPending(context, x.icon, x.pending, x.size)}
+				${when(x => x.label, html`<span class="text" role="presentation">${(x) => x.label}</span>`)}
```
Comment (rachelbt) [inline libs/components/src/lib/button/button.template.ts:96]: role="presentation" - why is this needed?
simple span do not have a role anyway.

**Review by rachelbt — COMMENTED** [review]
nice!
I don't think it is such an accessibility issue the group with when there's only icon.
I like the aria-hidden that you added - I guess we need to add this fix to split vbutton as well and to nav-disclosure too.

File: libs/components/src/lib/button/button.template.ts (hunk 1 above)
Comment (RichardHelm) [reply libs/components/src/lib/button/button.template.ts:96]: Without it, it still gets announced as group in VoiceOver. Maybe that is an issue in VoiceOver.

Comment (rachelbt) [reply libs/components/src/lib/button/button.template.ts:96]: I wonder then why adding the role presentation to the host didn't solved the group issue with the only icon. But I think we can leave it.

**Review by rachelbt — APPROVED** [review]
LGTM
👍

File: libs/components/src/lib/button/button.template.ts (hunk 1 above)
Comment (RichardHelm) [reply libs/components/src/lib/button/button.template.ts:96]: That one is actually working as intended, unfortunately: https://www.w3.org/WAI/ARIA/apg/practices/hiding-semantics/#presentation_role_ignored
